### ISSUE TITLE
Smaller epub size

### ISSUE THREAD

**Issue by Stoneguard001** [hendricius/the-sourdough-framework#401]
I pulled down and built the book, but ran into an issue with the size of the epub in Amazon Kindle, whic allows a max size of 50MB.  I realize the black & white can be loaded, but prefer the color images. 

I have made some changes to the makefile to simply reduce the color images (copying the b&w make) and reduce the images so it all fits in under 30MB.  I just to be able to create a PR to present the changes. They are by no means perfect; I am not a LaTex developer, so I don't know its inner working so well and am completely open to reworking my changes if there is a more efficient way of doing the image reduction.

**Comment by cedounet** [comment]
Great, go ahead and submit a PR.  

I thought kindle were black and white though?  The colour was more meant for reading on iPad or a tablet where size limitations do not really exist.

**Comment by Stoneguard001** [comment]
Clarification: I am using the Kindle reader on my cellphone. This shows color. Working on sorting out my local GitHub Identities between work and personal at the moment and will hopefully get a PR up soon :)

**Comment by cedounet** [comment]
If that helps: https://www.benji.dog/articles/git-config/ to manage different identities.

**Comment by cedounet** [comment]
Merged changes in https://github.com/hendricius/the-sourdough-framework/pull/402 but that is not published automatically.

@Stoneguard001 @hendricius  let's have discussion on what we want long term:

1) Are we replacing some images?
2) Do we need to change size for ebook or just reduce quality?
3) Reducing quality on png not automatically a good idea

Personally i don't mind a large pdf (within reasons) but if kindle apps have a hard limit then ebooks should comply.

**Comment by Stoneguard001** [comment]
My ultimate goal was to get the size of the epub for Amazon's [Send To Kindle](https://www.amazon.com/sendtokindle) function under their 50MB epub limit without sacrificing a lot of image quality.  The recently merged process reduces the size from 67MB to ~28MB.  I am all for skipping any unneeded changes, but ultimately I would like to keep it under that 50MB limitation, though I am sure there are other ways to convert to mobi.

**Comment by hendricius** [comment]
Makes sense with the 50mb limit for kindle devices. I am sure that other images can be optimised/made smaller too to reduce file sizes. I will keep that in mind when I update some of the book's images to match the hardcover version.

**Comment by cedounet** [comment]
Thanks  both,

@hendricius  when do you think you will change the pictures?  and do you think this will solve the size problem by itself?

If you are not really sure when, I think we should replace the epub by the low_res version and see if people notice the difference.

Steps for that would be:
- copy .png instead of reducing quality
- have release target to use the low res epub when doing the copy
-  Add a test to make sure this never goes > 50 MB in the makefile.

@Stoneguard001 if you are happy to do that great, we can work on it together if not I will give it a go after my work on new plots.

**Comment by Stoneguard001** [comment]
@cedounet that's sounds like a solid plan to me. I will wait on a response from @hendricius before moving forward with anything, but I think the first two bullet points should be relatively simple and I can do those in a new PR.

**Comment by hendricius** [comment]
> Thanks both,
> 
> @hendricius when do you think you will change the pictures? and do you think this will solve the size problem by itself?
> 
> If you are not really sure when, I think we should replace the epub by the low_res version and see if people notice the difference.
> 
> Steps for that would be:
> 
> * copy .png instead of reducing quality
> * have release target to use the low res epub when doing the copy
> * Add a test to make sure this never goes > 50 MB in the makefile.
> 
> @Stoneguard001 if you are happy to do that great, we can work on it together if not I will give it a go after my work on new plots.

Nope - it will not likely fix the issue. It might even make the book larger. So I propose we go ahead with the plan and then having the 50mb file size limit will be a great check for me too to keep things balanced with quality and file size.

Thanks for the great changes!

**Comment by cedounet** [comment]
Bad news @Stoneguard001 it failed on CI: https://github.com/hendricius/the-sourdough-framework/actions/runs/12561596069 (if you have access...)

If you can fix quickly great (will have a look myself but not for long) if not I will revert your commit.

**Comment by cedounet** [comment]
Looks like this is the culprit.. make does not know how to make them

``make: *** No rule to make target 'low-res-book-epub/OEBPS/images/external/wheat-kernel-overview.png', needed by ``'epub/low_res_book.epub'.  Stop.
``make: *** Waiting for unfinished jobs....

**Comment by Stoneguard001** [comment]
Of. I am out and about right now, but will look at it in an hour. Go ahead and revert for now, sorry I'm unavailable at the moment!

**Comment by cedounet** [comment]
No worries... I might have a fix... testing right now and will push in a few minutes if it works.

PS: Always a good idea to compile with -j locally might have caught that bug.

**Comment by cedounet** [comment]
Fix worked... not sure I am super proud of it but at least build isn't broken anymore.

Thanks a lot for the PR and reworks.

**Comment by Stoneguard001** [comment]
Awesome, thanks for fixing it. I'll take your testing suggestion for future builds!
